src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1319,7 +1319,12 @@ smart_objects::SmartObjectList MessageHelper::CreateAddCommandRequestToHMI(
 
       smart_objects::SmartObject msg_params =
           smart_objects::SmartObject(smart_objects::SmartType_Map);
-      msg_params[strings::cmd_id] = i->first;
+
+      if ((*i->second).keyExists(strings::cmd_id)) {
```

Comment:
@APipko `cmd_id` is a mandatory parameter for AddCommand, so if it is missing then whole request will be invalid anyway. I suggest do not return such request from this function at all.